### PR TITLE
[MM-30447] - Add create direct message option to category menu

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
This PR adds the create direct message option to the main category menu. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30447

### CODE REVIEW

File: components/sidebar/sidebar_category/sidebar_category_menu/sidebar_category_menu.tsx [hunk 1]
```diff
@@ -126,13 +127,22 @@ class SidebarCategoryMenu extends React.PureComponent<Props, State> {
                 />
             );
         }
+        const createDirectMessage = (
```
Comment (devinbinnie) [inline components/sidebar/sidebar_category/sidebar_category_menu/sidebar_category_menu.tsx:130]: Do we want this to show up on every category menu or just the Direct Messages category?

**Review by devinbinnie — APPROVED** [review]

File: components/sidebar/sidebar_category/sidebar_category_menu/sidebar_category_menu.tsx (hunk 1 above)
Comment (nevyangelova) [reply components/sidebar/sidebar_category/sidebar_category_menu/sidebar_category_menu.tsx:130]: from what I understand from the ticket and spec it's all categories. @matthewbirtch can you confirm?

**Comment by jgilliam17** [comment]
@nevyangelova I don't see the changes on the test server. Can you please check - thanks 🙂

**Comment by nevyangelova** [comment]
@jgilliam17 they are showing up for me

**Comment by jgilliam17** [comment]
Ah sorry about that - user error 😅

**Review by hmhealey — CHANGES_REQUESTED** [review]
Sorry, there's a similar issue here to the other PR. This ticket was  to add it to the plus button menu at the top, not the menu for the individual categories. We really need better names for these 😛

**Comment by jgilliam17** [comment]
Thanks @nevyangelova 
Works great. One suggestion - change text to "Open a Direct Message" with all words capitalized

**Review by jgilliam17 — APPROVED** [review]
Thank you @nevyangelova 
Tested, looks good to merge.
- Verified "Open a Direct Message" option has been added to sidebar main + menu, behaves the same as DM +.

File: components/sidebar/sidebar_category/sidebar_category_menu/sidebar_category_menu.tsx (hunk 1 above)
Comment (matthewbirtch) [reply components/sidebar/sidebar_category/sidebar_category_menu/sidebar_category_menu.tsx]: I don't think there was any intention to add a direct message option to all category menus - just the DM category. Do you have a design reference where this is specified? I don't recall this recommendation.

There may be confusion. The intent was to add the option to create a new direct message from the top main `+` button, but not from every category.
image

Comment (nevyangelova) [reply components/sidebar/sidebar_category/sidebar_category_menu/sidebar_category_menu.tsx]: @matthewbirtch there was a confusion but Harrison cleared it out for me. I added the button to the top menu as specified. You can review on the test server.

Comment (matthewbirtch) [reply components/sidebar/sidebar_category/sidebar_category_menu/sidebar_category_menu.tsx]: Ok, thanks for clearing that up! Looks good to me

**Review by matthewbirtch — APPROVED** [review]
Looks good!

**Review by hmhealey — APPROVED** [review]

**Pull request merged**